lib/matplotlib/colors.py

```diff
@@ -958,6 +958,677 @@ def scaled(self):
+                 refpoints_data=[None],
+                 refpoints_cm=[None],
+                 **normalize_kw):
+        """
+        Specify a series of functions, as well as intervals to map the data
+        space into `[0,1]`. Each individual function may not diverge in the
+        [0,1] interval, as they will be normalized as
+        fnorm(x)=(f(x)-f(0))/(f(1)-f(0)), to guarantee that fnorm(0)=0 and
+        fnorm(1)=1. Then each function will be transformed to map a different
+        data range [d0, d1] into colormap ranges [cm0, cm1] as
+        ftrans=fnorm((x-d0)/(d1-d0))*(cm1-cm0)+cm0.
+
+        Parameters
+        ----------
+        flist : list of callable or strings
+            List of functions to be used for each of the intervals.
+            Each of the elements must meet the same requirements as the
+            parameter `f` from `FuncNorm`.
+        finvlist : list of callable or strings, optional
+            List of the inverse functions corresponding to each function in
+            `flist`. Each of the elements must meet the same requirements as
+            the parameter `finv` from `FuncNorm`. None may be provided as
+            inverse for the functions that were specified as a string in
+            `flist`. It must satisfy `len(flist)==len(finvlist)`.
+        refpoints_cm, refpoints_data : list or array of scalars
+            Reference points for the colorbar ranges which will go as
+            `[0., refpoints_cm[0]]`,... ,
+            `[refpoints_cm[i], refpoints_cm[i+1]]`,
+            `[refpoints_cm[-1], 0.]`, and for the data
+            ranges which will go as `[self.vmin, refpoints_data[0]]`,... ,
+            `[refpoints_data[i], refpoints_data[i+1]]`,
+            `[refpoints_cm[-1], self.vmax]`
+            It must satisfy
+            `len(flist)==len(refpoints_cm)+1==len(refpoints_data)+1`.
+            `refpoints_data` must consist of increasing values
+            in the (vmin, vmax) range.
+            `refpoints_cm` must consist of increasing values be in
+            the (0.0, 1.0) range.
+            The final normalization will meet:
+            `norm(refpoints_data[i])==refpoints_cm[i]`.
+        normalize_kw : dict, optional
+            Dict with keywords (`vmin`,`vmax`,`clip`) passed
+            to `matplotlib.colors.Normalize`.
+
+        Examples
+        --------
+        Obtaining a normalization to amplify features near both -0.4 and
+        1.2 using four intervals:
+
+        >>> import matplotlib.colors as colors
+        >>> norm = colors.PiecewiseNorm(flist=['cubic', 'crt', 'cubic', 'crt'],
+        >>>                             refpoints_cm=[0.25, 0.5, 0.75],
+        >>>                             refpoints_data=[-0.4, 1, 1.2])
+
+        """
+
+        if finvlist is None:
+            finvlist = [None] * len(flist)
+
+        if len(flist) != len(finvlist):
+            raise ValueError("len(flist) must be equal than len(finvlist)")
```

Comment:
Also,  I'm on the fence,  but what about factoring out all this validation into a check-in put private function just so it doesn't clutter reading the function.